public/app/plugins/datasource/prometheus/language_provider.ts

```diff
@@ -501,19 +503,74 @@ export default class PromQlLanguageProvider extends LanguageProvider {
     return [];
   }
 
+  /**
+   * Gets series values
+   * Function to replace old getSeries calls in a way that will provide faster endpoints for new prometheus instances,
+   * while maintaining backward compatability
+   * @param labelName
+   * @param selector
+   */
+  getSeriesValues = async (labelName: string, selector: string): Promise<string[]> => {
+    if (!this.datasource.hasLabelsMatchAPISupport()) {
+      const data = await this.getSeries(selector);
+      return data[labelName] ?? [];
+    }
+    return await this.fetchSeriesValuesWithMatch(labelName, selector);
+  };
+
   /**
    * Fetches all values for a label, with optional match[]
    * @param name
    * @param match
    */
-  fetchSeriesValues = async (name: string, match?: string): Promise<string[]> => {
+  fetchSeriesValuesWithMatch = async (name: string, match?: string): Promise<string[]> => {
     const interpolatedName = name ? this.datasource.interpolateString(name) : null;
     const range = this.datasource.getTimeRangeParams();
     const urlParams = {
       ...range,
-      ...(interpolatedName && { 'match[]': match }),
+      ...(match && { 'match[]': match }),
     };
-    return await this.request(`/api/v1/label/${interpolatedName}/values`, [], urlParams);
+
+    const cacheParams = new URLSearchParams({
+      'match[]': interpolatedName ?? '',
+      start: roundSecToMin(parseInt(range.start, 10)).toString(),
+      end: roundSecToMin(parseInt(range.end, 10)).toString(),
+      name: name,
+    });
+
+    const cacheKey = `/api/v1/label/?${cacheParams.toString()}/values`;
```

Comment:
This is a bit of a regression risk, and will introduce cacheing of this query (rounding up to nearest minute) for both query builder and code editor.